docs/examples/nginx-services/README.md

```diff
@@ -0,0 +1,88 @@
+# Interlock with Docker Services
+Starting in Docker 1.12 there is support for Services.  In this example, we
+will use Services to run Interlock and the Proxy.
+
+Note: if you have not initialized the Swarm, make sure to init it with `docker swarm init`
+
+# Nginx
+This will create a global Nginx service that will run on every node and
+publish port 80:
+
+```
+docker service create \
+    --name interlock-nginx \
+    --publish 80:80 \
+    --mode global \
+    --label interlock.ext.name=nginx \
+    nginx \
+        nginx -g "daemon off;" -c /etc/nginx/nginx.conf
+```
+
+You should now be able to visit `http://<node-ip>` and see the Nginx welcome
+page.
+
+# Interlock
+We will now start a global Interlock service that will run on every node:
+
+First, we need an Interlock configuration.  Create a file called `config.toml`
+with the content:
+
+```
+ListenAddr = ":8080"
+DockerURL = "unix:///var/run/docker.sock"
+PollInterval = "2s"
+
+[[Extensions]]
+Name = "nginx"
+ConfigPath = "/etc/nginx/nginx.conf"
+PidPath = "/var/run/nginx.pid"
+TemplatePath = ""
+MaxConn = 1024
+Port = 80
+```
+
+Now create the Interlock service:
+
+```
+docker service create \
+    --mode global \
+    --name interlock \
+    --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock,writable=true \
```

Comment:
Thanks. I'll re-vendor and update.

On Jul 31, 2016 05:51, "Luc Vieillescazes" <notifications@github.com> wrote:

> In docs/examples/nginx-services/README.md
> <https://github.com/ehazlett/interlock/pull/186#discussion_r72901527>:
>
> > +[[Extensions]]
> > +Name = "nginx"
> > +ConfigPath = "/etc/nginx/nginx.conf"
> > +PidPath = "/var/run/nginx.pid"
> > +TemplatePath = ""
> > +MaxConn = 1024
> > +Port = 80
> > +```
> > +
> > +Now create the Interlock service:
> > +
> > +```
> > +docker service create \
> > +    --mode global \
> > +    --name interlock \
> > +    --mount type=bind,source=/var/run/docker.sock,target=/var/run/docker.sock,writable=true \
>
> According to docker/docker#24053
> <https://github.com/docker/docker/pull/24053>
> writable=true is the default behavior and the flag has been removed
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/ehazlett/interlock/pull/186/files/97977eb4a071a8c515ac030999a90c798d6477d1#r72901527>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAP6ImaphFDoOaO14zzA9lOOZCetHiXhks5qbHAzgaJpZM4JIiDb>
> .
>